src/core/server/external_url/config.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { TypeOf, schema } from '@kbn/config-schema';
+import { IExternalUrlPolicy } from '.';
+
+/**
+ * @internal
+ */
+export type ExternalUrlConfigType = TypeOf<typeof config.schema>;
+
+const allowSchema = schema.boolean();
+
+const hostSchema = schema.string();
+
+const protocolSchema = schema.string({
+  validate: (value) => {
+    if (value.indexOf('/') >= 0) {
+      throw new Error(`protocols should not include slashes`);
+    }
+  },
+});
```

Comment:
Nit: we could be a bit more strict in our validation:

https://tools.ietf.org/html/rfc3986#section-3.1

> Scheme names consist of a sequence of characters beginning with a
   letter and followed by any combination of letters, digits, plus
   ("+"), period ("."), or hyphen ("-").  Although schemes are case-
   insensitive, the canonical form is lowercase and documents that
   specify schemes must do so with lowercase letters.  An implementation
   should accept uppercase letters as equivalent to lowercase in scheme
   names (e.g., allow "HTTP" as well as "http") for the sake of
   robustness but should only produce lowercase scheme names for
   consistency.
>
>       scheme      = ALPHA *( ALPHA / DIGIT / "+" / "-" / "." )

Reply:
Sure, updated in [`da2bf28` (#81234)](https://github.com/elastic/kibana/pull/81234/commits/da2bf289bf86bd4f56239329c3df80c011b024e0)